public/whiteboard/js/main.js

```diff
@@ -1,14 +1,103 @@
 /* global */
+let board
+let img
+let brush
+let cnv
+let bSize
+let colorPicker
+let currColor
+let sizeSlider
+let saveButton
+
+/* Socket info to send;
+    mouse position
+    current canvas (as png? encoded 64bit?)
+     */
+
+function createBrush(hex){
```

Comment:
creating a function to contain all the logic corresponding to creating a brush is great "modular" thinking! but a brush isn't just something that gets "created", it's also got "properties" that get "updated", like it's color (which u currently have stored in a global variable called `currColor`) && it's size (currently stored in `bSize`), which means we can approach modularizing/abstracting this code a little differently. specifically using OOP ([object oriented programming](https://en.wikipedia.org/wiki/Object-oriented_programming)). Here's an example of a potential "Brush" [class](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) which can be used to instantiate as many diff brushes as u need, each w/their own size, color, img && any other properties u decide to add to it later on. u can also add as many methods (like the `draw()` method i made) as u need as the project develops.

```js
class Brush {
  constructor (img) {
    // list bursh properties here
    this.size = 50
    this.color = '#000'
    this.img = img
  }

  draw (x, y) {
    x = x - this.size / 2
    y = y - this.size / 2
    image(this.img, x, y, this.size, this.size)
  }
}
```

it's convention to name a class w/a capital letter. && it's best practice to save them in their own file (i'd put it in `public/whiteboard/js/Brush.js`, unless we think other groups might benefit from this file as well, in which case i would put it in `public/js/Brush.js`) so that they might be more easily re-used in other parts of the app (++ in a future project u work on that might need similar functionality, it'll be easier to copy+paste this self-contained logic into that project).